src/OrbitGl/TrackManager.cpp

```diff
@@ -60,7 +60,11 @@ TrackManager::TrackManager(TrackContainer* track_container, TimelineInfoInterfac
 }
 
 std::vector<Track*> TrackManager::GetAllTracks() const {
-  std::lock_guard<std::recursive_mutex> lock(mutex_);
+  absl::MutexLock lock(&mutex_);
```

Comment:
You might want to consider using a `ReaderLock` here and at all the other "getters". I am interested in other thoughts here.

Reply:
(Note: If you move to read and write locks, you might need different annotations as mentioned in the other comment `ABSL_EXCLUSIVE_LOCKS_REQUIRED` refers to the required write locks, while `ABSL_SHARED_LOCKS_REQUIRED` refers to the required read locks.)

Reply:
it makes sense to me, thanks!